apps/server/src/modules/oauth/controller/oauth-sso.controller.ts

```diff
@@ -57,6 +59,34 @@ export class OauthSSOController {
 		res.redirect(errorRedirect.toString());
 	}
 
+	private migrationErrorHandler(
+		error: unknown,
+		session: ISession,
+		res: Response,
+		sourceSystemId: string,
+		targetSystemId: string
+	) {
+		this.logger.error(error);
+		const migrationError: OAuthMigrationError =
+			error instanceof OAuthMigrationError ? error : new OAuthMigrationError();
+
+		session.destroy((err) => {
+			this.logger.log(err);
+		});
+
+		const errorRedirect: URL = new URL('/migration/error', this.clientUrl);
+
+		errorRedirect.searchParams.append('sourceSystem', sourceSystemId);
```

Comment:
Line 77-79 to 85 is read like a mapper. Can it be outsource?

Reply:
This will be moved to client in the next two sprints :
https://ticketsystem.dbildungscloud.de/secure/RapidBoard.jspa?rapidView=162&view=planning&selectedIssue=N21-749&issueLimit=100

to remove technical debt. I want to avoid generate more dept in this case.. is that ok for now?